packages/sdk/src/swap/sdk.ts

```diff
@@ -285,20 +285,33 @@ export class SwapSDK {
     txOpts: TransactionOptions & { signer?: Signer } = {},
   ): Promise<TransactionHash> {
     const { srcChain, srcAsset, amount } = params;
+    const { network } = this.options;
+    const internalAsset = getInternalAsset({ chain: srcChain, asset: srcAsset });
 
     const { signer: optsSigner, ...remainingTxOpts } = txOpts;
     const signer = optsSigner ?? this.options.signer;
     assert(signer, 'No signer provided');
 
     await this.validateSwapAmount({ chain: srcChain, asset: srcAsset }, BigInt(amount));
 
+    if (!isGasAsset(internalAsset)) {
+      const allowanceInfo = await checkVaultAllowance(params, {
+        signer,
+        network,
+      });
+
+      if (!allowanceInfo.hasSufficientAllowance) {
+        await this.approveVault(params, txOpts);
+      }
```

Comment:
approve vault already checks for sufficient allowance
```suggestion
      await this.approveVault(params, txOpts);
```